### ISSUE TITLE
question sur moissonnage dcat-data.gouv

### ISSUE THREAD

**Issue by mkazmiers** [etalab/data.gouv.fr#95]
Bonjour,

notre prestataire a branché le plugin ckanext-dcat sur le ckan de dev de DATASUD.
J'ai créé une orga "ORGANISATION_GEO" sur next.data.gouv.fr et voulais créer un moissonneur. 

Je n'ai pas trouvé de doc spécifique ici : https://doc.data.gouv.fr/faq/developpeur/#section-moissonage
Je vous pose donc la question ici : que dois-je indiquer comme endpoint pour que les données de mon "ORGANISATION_GEO" de trouver-rec.datasud.fr soit moissonnée vers "ORGANISATION_GEO" de next.data.gouv ?

NB : c'est pour l'instant une question théorique, car notre infra de recette filtre les IP entrantes ; est-ce possible à ce titre de disposer de l'IP publique de next.data.gouv (c'est bien celle que je récupère par ping de votre url ou en existe-t-il plusieurs ?...ou est-ce bien plus compliqué ?) ?

Mathieu

**Comment by noirbizarre** [comment]
Bonjour,

Vous devez indiquer l'URL de votre catalogue en RDF/DCAT (quelque soit le format, JSON-LD, XML, TTL...).

Par contre, deux questions me viennent à l'esprit ?
- pourquoi utiliser le moissonneur DCAT et non le moissonneur CKAN pour moissonner votre portail en CKAN ?
- pourquoi appliquer un filtrage par IP sur un portail open data ? Qu'est-ce qui justifie une telle restriction ?

**Comment by mkazmiers** [comment]
OK. Donc en théorie, trouver-rec.datasud.fr/catalog.xml fonctionne ? Ou faut-il indiquer l'organisation en particulier ? On ne veut pas que l'ORGANISATION_GEO rapatrie les 500 jeux de données de toutes les organisations du CKAN, mais juste les 3 publiées pour l'ORGANISATION_GEO...j'espère que c'est plus clair ? 

* **pourquoi utiliser le moissonneur DCAT et non le moissonneur CKAN pour moissonner votre portail en CKAN ?**
Après des échanges avec qqn chez vous, nous avions cru comprendre que le moissonnage via CKAN ne nous permettait pas de filtrer le moissonnage sur 1 organisation + le modèle de données CKAN n'est pas standard et qu'on prend le risque d'avoir des champs manquants. On nous a donc suggéré de passer par DCAT, plus standardisé, ce qu'on a donc poussé.

* **pourquoi appliquer un filtrage par IP sur un portail open data ? Qu'est-ce qui justifie une telle restriction ?**
c'est juste que c'est l'infrastructure de recette et qu'elle évolue constamment ces jours-ci. c'est un choix de de façon de dev...L'infrastructure en production n'a pas de restriction, donc il ne sera pas question d'IP.

**Comment by mkazmiers** [comment]
OK, le plugin ckanext-dcat de DATASUD a été augmenté de la possibilité de filtrer sur une organisation...sauf si il y a une façon de faire standard que nous n'avons pas identifié à ce stade. On va tester ça : https://trouver-rec.datasud.fr/catalog.xml?organization=organisation_geo

On a récupéré l'ip public qu'on ping depuis next.data.gouv, j'espère que ce sera suffisant pour que vous accédiez à la recette.

**Comment by abulte** [comment]
Pour info le moissonneur CKAN de data.gouv.fr supporte depuis une ou deux semaines le filtrage, notamment par organisation : https://github.com/opendatateam/udata/pull/1812.

**Comment by abulte** [comment]
Vous pouvez voir le résultat « à blanc » du moissonneur sans qu’il soit validé en cliquant sur « Prévisualisation » sur la page du moissonneur. En l’occurence il semble tourner dans le vide, ce qui ressemble à un problème d’accès à votre plateforme. Je ne peux pas vous répondre sur l’IP de next.data.gouv.fr mais vous pouvez peut-être la voir dans vos logs quand vous lancez une prévisualisation.

**Comment by mkazmiers** [comment]
Effectivement, la prévisu tourne dans le vide sur le endpoint filtré https://../catalog.xml?organization=organisation_geo 

Comme la prévisu ne tourne pas, je n'ai pas accès au log, je ne peux pas vérifier que next.data.gouv.fr accède bien au endpoint. On a ouvert l'IP que l'on ping, mais ça ne marche pas mieux. Est-il possible de vous contacter en direct par mp ou mail pour déterminer quel est l'ip sortante du moissonneur ?

**Comment by abulte** [comment]
Vous pouvez nous contacter sur info@data.gouv.fr. Je vais me renseigner sur l’IP de sortie.
